### PR TITLE
Prepare import command for future extensions

### PR DESCRIPTION
**Pull request by bhiemenz**
Please note: I revert my changes in the Georocket class as ykammeyer already made a pull request for these changes.

### CODE REVIEW

File: georocket-cli/src/main/java/io/georocket/GeoRocketCli.java [hunk 1]
```diff
@@ -159,6 +147,25 @@ public static void main(String[] args) throws IOException {
     }
   }
 
+  /**
+   * Setup GeoRocket Cli.
+   */
+  protected static void setup() {
```
Comment (michel-kraemer) [inline georocket-cli/src/main/java/io/georocket/GeoRocketCli.java]: static method cannot be overridden in Java, so why this change? Wouldn't it make more sense to reduce the `main` method to only two lines?

```
GeoRocketCli cli = new GeoRocketCli();
cli.run(args);
```

Or something similar?

File: georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java [hunk 2]
```diff
@@ -45,8 +45,8 @@
  */
 public class ImportCommand extends AbstractGeoRocketCommand {
   private List<String> patterns;
```
Comment (michel-kraemer) [inline georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java]: Should `patterns` be `protected` as well?

File: georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java [hunk 3]
```diff
@@ -212,20 +212,8 @@ private void doImport(Queue<String> files, GeoRocketClient client,
...
-      .map(f -> {
-        System.out.print("Importing " + Paths.get(path).getFileName() + " ... ");
-        return f;
-      })
-      // get file size
-      .flatMap(f -> fs.propsObservable(path).map(props -> Pair.of(f, props.size())))
-      // import file
-      .flatMap(f -> importFile((AsyncFile)f.getLeft().getDelegate(), f.getRight(), client))
+
+    prepareFile(path, client, vertx)
```
Comment (michel-kraemer) [inline georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java]: It seems `prepareFile` really imports the file. The name is misleading.

File: georocket-client-api/src/main/java/io/georocket/client/StoreClient.java [hunk 4]
```diff
@@ -12,6 +12,7 @@
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
```
Comment (michel-kraemer) [inline georocket-client-api/src/main/java/io/georocket/client/StoreClient.java]: Unused import?

File: georocket-client-api/src/main/java/io/georocket/client/StoreClient.java [hunk 5]
```diff
@@ -170,28 +199,32 @@ protected String prepareQuery(String query, String layer) {
...
+   * special options, the option setting is outsourced to this
+   * method.
+   *
+   * @param request request the options are created for
+   * @param size size of the data to be sent in bytes (optional)
+   * @param handler a handler that will be called when the data has been
+   * imported by the GeoRocket server
+   * @return same {@link HttpClientRequest} as given as parameter but with options set
+   */
+  protected HttpClientRequest requestOptions(HttpClientRequest request, Optional<Long> size, Handler<AsyncResult<Void>> handler) {
```
Comment (michel-kraemer) [inline georocket-client-api/src/main/java/io/georocket/client/StoreClient.java]: Line is very long. It almost doesn't fit here in the diff view. Please wrap it like in `startImport`.

**Comment by michel-kraemer** [comment]
Almost good. Thanks a lot. Can you please fix these minor issues?

File: georocket-client-api/src/main/java/io/georocket/client/StoreClient.java (hunk 5 above)
Comment (bhiemenz) [inline georocket-client-api/src/main/java/io/georocket/client/StoreClient.java]: Done, thanks.

File: georocket-client-api/src/main/java/io/georocket/client/StoreClient.java (hunk 4 above)
Comment (bhiemenz) [inline georocket-client-api/src/main/java/io/georocket/client/StoreClient.java]: Done, thanks.

File: georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java (hunk 3 above)
Comment (bhiemenz) [inline georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java]: Done, thanks.

File: georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java (hunk 2 above)
Comment (bhiemenz) [inline georocket-cli/src/main/java/io/georocket/commands/ImportCommand.java]: No, as the pattern does not have to be handled by the inherited class

File: georocket-cli/src/main/java/io/georocket/GeoRocketCli.java (hunk 1 above)
Comment (bhiemenz) [inline georocket-cli/src/main/java/io/georocket/GeoRocketCli.java]: I do not overwrite this method, I just call it. Nevertheless, I think your way is better coding. Thanks.
